aws-samples/amazon-transcribe-post-call-analytics
# Feature/clm

**Description of changes:**

Updated use of the two API calls to start Transcribe jobs to use the recent changes to specify a _Custom Vocabulary_ and a _Vocabulary Filter_ at the same time as using _Language ID_.  This allowed us to remove the previous Lambda functions and Step Function steps that created a 30-second clip for language ID purposes before calling Transcribe again with a specified language and its vocab and filter files.  As it is a new API call signature this required an update of the boto3 version.

Also fixed a bug whereby the start/end times of words for TCA were accidentally set to start/start.

**Testing**

The following tests were carried out:

1. Set `TranscribeApiMode` configuration to `analytics`
2. Set `TranscribeLanguages` configuration `en-US | fr-FR`
3. Define a `VocabularyName` or `VocabFilterName` for `en-US`, and set them up in Transcribe
4. Drop any of the default files into the input bucket and verify that (a) it processes, and (b) the API integration panel of that job indicates that it was a language identification job
5. Set `TranscribeLanguages` configuration `en-US`
6. Drop any of the default files into the input bucket and verify that (a) it processes, and (b) the API integration panel of that job indicates that it was a specific-language job
7. Set `TranscribeApiMode` configuration to `standard`
8. Repeat steps 2 to 7

All files were processed correctly, and the Transcribe console correctly indicated if it was a specified language job or a language identification job

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
